File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        pattern = '[^A-Za-z0-9]'
        phrase = re.sub(pattern, '', s).lower()
        
        l, r = 0, len(phrase) - 1
        while l <= r:
            if phrase[l] == phrase[r]:
                l += 1
                r -= 1
            else:
                return False

        return True
        

File: 0125-valid-palindrome/0125-valid-palindrome.go
import "regexp"
func isPalindrome(s string) bool {
    pattern := regexp.MustCompile("[^A-Za-z0-9]")
    phrase := strings.ToLower(pattern.ReplaceAllString(s, ""))

    l, r := 0, len(phrase) - 1
    fmt.Print(phrase)
    for l < r {
        if phrase[l] != phrase[r]{
            return false
        }
        l++
        r--
    }

    return true
}